scripts/lib/CIME/case.py

```diff
@@ -951,15 +953,26 @@ def create_caseroot(self, clone=False):
             self._create_caseroot_sourcemods()
         self._create_caseroot_tools()
 
+        # Apply user_mods if part of compset
+        if self._user_mods is not None:
+            self._user_mods = self.get_resolved_value(self._user_mods)
```

Comment:
Is there a reason that get_resolved_value is done here, at the last minute, rather than being done when this element is first retrieved from the xml file? It seems like the latter would be less error-prone as the code evolves. But I don't know what the standard convention is in this respect.